compiler/noirc_evaluator/src/ssa/opt/check_u128_mul_overflow.rs

```diff
@@ -60,6 +61,21 @@ impl Function {
     }
 }
 
+/// MAX_NON_OVERFLOWING_CONST_ARG is expected to be [p/U],
+/// where U=U128::max() and p is the field modulus.
+///
+/// Then x<[p/u]<=p/U, so x*U<p
```

Comment:
Small nit, max_non_overflowing_const_arg is now used as a 'maximum', i.e for x up to this value, included.
So we need in fact [p/u]<p/U, but this is fine because p is prime so it cannot be a multiple of U (and U is not prime).
So the comment should rather be: x<=[p/U]< p/U, so x*U<p